wa/workloads/uibenchjanktests/__init__.py

```diff
@@ -0,0 +1,133 @@
+                  description='Extra arguments for am instrument.'),
+        Parameter('no_hidden_api_checks', kind=bool, default=False,
+                  description='Disables restrictions on the use of hidden '
+                  'APIs.'),
+    ]
+
+    def __init__(self, target, **kwargs):
+        super(Uibenchjanktests, self).__init__(target, **kwargs)
+
+        if 'iterations' not in self.instrument_args:
+            self.instrument_args['iterations'] = 1
+
+        self.dut_apk = PackageHandler(
+            self,
+            package_name=self._DUT_PACKAGE,
+            variant=self.variant,
+            strict=self.strict,
+            version=self.version,
+            force_install=self.force_install,
+            install_timeout=self.install_timeout,
+            uninstall=self.uninstall,
+            exact_abi=self.exact_abi,
+            prefer_host_package=self.prefer_host_package,
+            clear_data_on_reset=self.clear_data_on_reset)
+        self.apk = TestPackageHandler(
+            self,
+            package_name=self.package_name,
+            variant=self.variant,
+            strict=self.strict,
+            version=self.version,
+            force_install=self.force_install,
+            install_timeout=self.install_timeout,
+            uninstall=self.uninstall,
+            exact_abi=self.exact_abi,
+            prefer_host_package=self.prefer_host_package,
+            clear_data_on_reset=self.clear_data_on_reset,
+            instrument_args=self.instrument_args,
+            raw_output=self.raw,
+            instrument_wait=self.wait,
+            no_hidden_api_checks=self.no_hidden_api_checks)
+
+    def initialize(self, context):
+        super(Uibenchjanktests, self).initialize(context)
+        self.dut_apk.initialize(context)
+        self.dut_apk.initialize_package(context)
+        if 'class' not in self.apk.args:
+            class_for_method = dict(self.apk.apk_info.methods)
+            class_for_method[None] = self._DEFAULT_CLASS
+            try:
+                method = class_for_method[self.test]
+            except KeyError as e:
+                msg = 'Unknown test "{}". Known tests:\n\t{}'
+                known_tests = '\n\t'.join(m for m in class_for_method.keys()
+                                          if m is not None
+                                          and m.startswith('test'))
+                raise ValueError(msg.format(e, known_tests))
+            klass = '{}.{}'.format(self.package_names[0], method)
+
+            if self.test:
+                klass += '#{}'.format(self.test)
+            self.apk.args['class'] = klass
+
+    _OUTPUT_SECTION_REGEX = re.compile(
+            r'(\s*INSTRUMENTATION_STATUS: gfx-[\w-]+=[-+\d.]+\n)+'
+            r'\s*INSTRUMENTATION_STATUS_CODE: (?P<code>-+\d)\n?', re.M)
+    _OUTPUT_GFXINFO_REGEX = re.compile(
+            r'INSTRUMENTATION_STATUS: (?P<name>[\w-]+)=(?P<value>[-+\d.]+)')
```

Comment:
Could these be placed at the beginning of the class with the others?

Reply:
Done